platform/wab/src/wab/client/components/sidebar/token-controls.tsx

```diff
@@ -282,53 +324,73 @@ export const TokenRow = observer(function TokenRow(props: {
   );
 });
 
-export const TokenFolderRow = observer(function TokenFolderRow(props: {
-  name: string;
-  path?: string;
-  tokenType: TokenType;
+interface TokenFolderRowProps {
+  folder: TokenFolder;
   matcher: Matcher;
-  groupSize: number;
   indentMultiplier: number;
   isOpen: boolean;
-}) {
-  const {
-    name,
-    path,
-    tokenType,
-    matcher,
-    groupSize,
-    indentMultiplier,
-    isOpen,
-  } = props;
-  const tokenControls = useTokenControls();
+  toggleExpand: () => void;
+}
+
+export const TokenFolderRow = observer(function TokenFolderRow(
+  props: TokenFolderRowProps
+) {
+  const { folder, matcher, indentMultiplier, isOpen, toggleExpand } = props;
   const studioCtx = useStudioCtx();
   const readOnly = isTokenReadOnly(studioCtx);
+  const [renaming, setRenaming] = React.useState(false);
 
   return (
     <RowGroup
       style={{
         height: TOKEN_ROW_HEIGHT,
         paddingLeft: getLeftPadding(indentMultiplier),
       }}
-      groupSize={groupSize}
+      groupSize={folder.count}
       isOpen={isOpen}
       showActions={!readOnly}
-      actions={
-        !readOnly && (
-          <IconButton
-            onClick={async (e) => {
-              if (isOpen) {
-                e.stopPropagation();
-              }
-              await tokenControls.onAdd(tokenType, path);
-            }}
-          >
-            <Icon icon={PlusIcon} />
-          </IconButton>
-        )
+      menu={
+        <FolderContextMenu
+          onAdd={async () => {
+            if (!isOpen) {
+              toggleExpand();
+            }
+            await folder.onAdd(folder.tokenType, folder.path);
+          }}
+          arenaDisplay={"token"}
+          onSelectRename={() => setRenaming(true)}
+          onDelete={() => folder.onDelete(folder)}
+        />
       }
+      actions={<div></div>}
```

Comment:
I had to do this to hide the `+` icon on the right, and couldn't find a better way. I'm not up to speed yet on how to work on generated Plasmic code.